### PR TITLE
Add functionality for node deletion

### PR DESCRIPTION
**Pull request by ProtoNXT**
A new button in the top right corner allows to toggle the delete mode. When enabled, a click on any node will delete that node along with all connected edges.

### CODE REVIEW

File: src/components/Canvas.vue [hunk 1]
```diff
@@ -101,6 +106,13 @@ export default {
         y: window.scrollY + this.scroll.y + 100
       }))
     },
+    removeNode (node, instance) {
+      if (this.deleteMode) {
+        let edgesCopy = Array.from(node.edges)
+        edgesCopy.forEach(this.removeEdge)
+        if (this.nodes.includes(instance)) this.nodes.splice(this.nodes.indexOf(instance), 1)
```
Comment (AKoll) [inline src/components/Canvas.vue]: This `instance` is equal to `node.instance`, so why have 2 separate parameters?

File: src/components/Node.vue [hunk 2]
```diff
@@ -108,6 +108,9 @@ export default {
     onDisconnected (edge) {
       if (this.edges.includes(edge)) this.edges.splice(this.edges.indexOf(edge), 1)
     },
+    onDelete () {
+      this.$emit('delete', this, this.instance)
```
Comment (AKoll) [inline src/components/Node.vue]: Regarding my comment in Canvas.vue, would just passing `this` not be cleaner here?

File: src/components/DeleteSwitch.vue [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <q-btn id="delete-switch"
+         color="negative"
```
Comment (AKoll) [inline src/components/DeleteSwitch.vue]: I feel like making the button not be red when the delete mode is off, would make it more apparent when the user is in delete mode.

Comment (ProtoNXT) [reply src/components/DeleteSwitch.vue]: This is true, I added an alternative color when the delete mode is off.

**Pull request merged**